var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -171,20 +158,34 @@ def configure(self):
                 fh.write('\n'.join(makefile_inc))
 
     def install(self, spec, prefix):
-        self.configure()
-
         targets = ['scotch']
         if '+mpi' in self.spec:
             targets.append('ptscotch')
 
-        if '+esmumps' in self.spec:
-            targets.append('esmumps')
-            if '+mpi' in self.spec:
-                targets.append('ptesmumps')
+        if self.spec.version >= Version('6.0.0'):
```

Comment:
I seem to remember having problems with the default targets for `esmumps` and `ptesmumps` in the `_esmumps` archives for the older versions (my builds would always fail on `ptesmumps`), but this may have had to do with parallel building.  I'm going to try out these builds with your changes and I'll let you know what I find.